apps/ewallet/lib/ewallet/validators/transaction_consumption_validator.ex

```diff
@@ -120,6 +117,45 @@ defmodule EWallet.TransactionConsumptionValidator do
     :ok
   end
 
+  defp validate_transaction_request(request, operation) do
+    expiration_reason =
+      request.expiration_reason && String.to_existing_atom(request.expiration_reason)
+
+    request
+    |> TransactionRequest.valid?()
+    |> Kernel.||(expiration_reason)
+    |> do_validate_transaction_request(operation)
+  end
+
+  defp do_validate_transaction_request(condition, :consume) do
+    case condition do
+      true ->
+        :ok
+
+      expiration_reason when not is_nil(expiration_reason) and is_atom(expiration_reason) ->
+        {:error, expiration_reason}
+
+      _ ->
+        {:error, :unknown_error}
+    end
+  end
+
+  defp do_validate_transaction_request(condition, :confirm) do
+    case condition do
+      true ->
+        :ok
+
+      :cancelled_transaction_request ->
+        :ok
+
+      expiration_reason when is_atom(expiration_reason) ->
```

Comment:
This one doesn't guard for `not is_nil(expiration_reason)` like the function above it?